lib/slack-ruby-bot/rspec/support/slack-ruby-bot/respond_with_slack_message.rb

```diff
@@ -1,27 +1,32 @@
 require 'rspec/expectations'
-
 RSpec::Matchers.define :respond_with_slack_message do |expected|
+  include SlackRubyBot::SpecHelpers
   match do |actual|
-    client = if respond_to?(:client)
-               send(:client)
-             else
-               SlackRubyBot::Client.new
-             end
+    client = respond_to?(:client) ? send(:client) : SlackRubyBot::Client.new
+    def client.test_messages
+      @test_received_messages
+    end
+
+    def client.say(options = {})
+      super
+      @test_received_messages = @test_received_messages.nil? ? '' : @test_received_messages
+      @test_received_messages += "#{options.inspect}\n"
+    end
 
     message_command = SlackRubyBot::Hooks::Message.new
     channel, user, message = parse(actual)
 
     allow(Giphy).to receive(:random) if defined?(Giphy)
 
-    expect(client).to receive(:message).with(channel: channel, text: expected)
+    allow(client).to receive(:message)
     message_command.call(client, Hashie::Mash.new(text: message, channel: channel, user: user))
+    @messages = client.test_messages
```

Comment:
Why not leave the old code with `expect`, and then you don't need the `failure_message` at all?